core/src/main/scala/za/co/absa/spline/harvester/converter/InputAttributeConverter.scala

```diff
@@ -16,24 +16,22 @@
 
 package za.co.absa.spline.harvester.converter
 
-import org.apache.spark.sql.catalyst.expressions.{Expression, Attribute => SparkAttribute}
-import za.co.absa.spline.harvester.builder.OperationNodeBuilder.OperationId
-import za.co.absa.spline.harvester.converter.ExpressionConverter.{ExpressionLike, toSplineAttrId, createExtra}
+import org.apache.spark.sql.catalyst.expressions.{Attribute => SparkAttribute, Expression => SparkExpression}
+import za.co.absa.spline.harvester.converter.ExpressionConverter.{ExpressionLike, createExtra}
 import za.co.absa.spline.producer.model.v1_1.Attribute
 
-class InputAttributeConverter(dataTypeConverter: DataTypeConverter) extends ExpressionConverter {
+class InputAttributeConverter(
```

Comment:
> Since we only concern about _input_ attributes (as the class name suggests). why not to use `AttributeReference` instead of an abstract 'Attribute'?

Why not program to interface/abstract class? Why make it more specific than necessary?